Add Order Management Feature and Update MyBatis Configuration

kebin0591 opened the pull request:
### 주문 관리 기능 추가 및 MyBatis 설정 개선

제가 담당한 주문 기능의 기초 파일들과 패키지 메서드 초안을 작성하였고, MyBatis 설정도 임시로 Mapper를 추가하였습니다~

 주요 변경 사항:

#### 1. 새로운 기능 추가
- **주문 관리 기능 구현**
  - **OrderController**: 주문 생성, 조회, 업데이트, 삭제 기능의 초안을 구현.
  - **OrderService**: 비즈니스 로직을 처리하고 주문 관련 작업을 효율적으로 관리하기위한 써-비스
  - **OrderMapper**: MyBatis 매퍼 인터페이스를 통한 데이터베이스와의 상호작용 기-능 .
  - **OrderDTO**: 그냥 DTO입니다
  - **OrderItemDTO**: 얘도 그냥 DTO입니다
  - **OrderView**: 뷰-입니다.
 
** 추가 **
1.   - **CartItem** 및 **CartItemDTO** 클래스를 추가하여 비즈니스 로직과 데이터 전송 객체를 분리
2.   - **OrderMainMenu**에 "주문하기" 메뉴 옵션을 추가
3.   - 사용자가 도서 코드를 검색하거나 전체 도서를 조회할 수 있습니다 (임시 구현).
4.   - 사용자가 도서를 장바구니에 담고, 수량을 지정하며, 결제 단계로 진행할 수 있음
5. - **OrderController**에 `createOrder` 메서드를 추가하여 주문 생성을 처리할 수 있도록 함
6. - **OrderService**를 업데이트하여 여러 주문 항목을 포함한 주문 생성을 지원(임시)
7. - **OrderDAO**에 `insertOrderItem` 메서드를 추가하여 데이터베이스에 주문 항목을 추가할 수 있도록 하였습니다.
8. - **OrderMapper** 인터페이스를 수정하여 주문 항목에 대한 SQL 쿼리를 매핑하였습니다.
9. - **Cart** 클래스를 개선하여 장바구니 항목을 관리하고, 데이터를 전송하기 위해 **CartDTO**로 변환할 수 있도록 하였습니다.
10. - **Input** 및 **Menu** 클래스에 유틸리티 메서드를 추가하여 사용자 상호작용을 개선하였습니다.

#### 2. MyBatis 설정
- **OrderMapper.xml**: 주문 관련 SQL 쿼리를 매핑.
- **mybatis-config.xml**: OrderMapper 매퍼를 등록.

#### 3. 패키지 구조 변경
- **패키지 구조 정리**: 모든 클래스와 파일을 `com.infinityiterators.bookwms` 패키지 내로 이동하여 프로젝트 구조를 일관성 만들었슘다,,

#### 4. 리팩토링
- utility를 사용한 리팩토링


추가적인 질문사항이나 수정사항 있으면 피드백 부탁드립니다~ 빠른 PR 리뷰 부탁드립니다~

kebin0591 committed:
Add Order Management Feature and Update MyBatis Configuration

주문 파트 주문 관리 기능 추가 및 MyBatis 설정 업데이트, 프로젝트 기초 structure 생성.

1. 새로운 기능 추가
   - 주문하기 기능을 위한 클래스 및 메서드 추가
     - OrderController: 주문 생성, 조회, 업데이트, 삭제 기능 구현
     - OrderService: 주문 관련 비즈니스 로직 구현
     - OrderMapper: MyBatis 매퍼 인터페이스 정의
     - OrderDTO: 주문 데이터 전송 객체 정의
     - OrderItemDTO: 주문 항목 데이터 전송 객체 정의
     - OrderView: 사용자 인터페이스 구현

2. MyBatis 설정
   - OrderMapper.xml 파일 추가: 주문 관련 SQL 쿼리 매핑
   - mybatis-config.xml 파일 수정: OrderMapper 매퍼 등록
3. 전체 프로젝트 토대 생성 완료

DokaDev commented on src/main/java/com/infinityiterators/bookwms/order/mapper/OrderMapper.xml:
이 파일은 java가 아닌 resource쪽에 패키지 구조 안에 배치되어야 하지 않나요?

kebin0591 replied:
아앗 앗 깜박했네요 수정하겠습니다~

kebin0591 replied:
> 이 파일은 java가 아닌 resource쪽에 패키지 구조 안에 배치되어야 하지 않나요?

파일 옮기기 했습니당~

ChaeUngLim replied:
<delete id="deleteOrder" parameterType="int"> 에서 parameterType="_int" 으로 변경하는게 좋을 것 같습니다

DokaDev reviewed (changes requested):
스키마 생성 쿼리도 /setup 디렉토리에 넣어주시고, 각 로직에 대한 단위 테스트도 진행하면 좋을 것 같습니다~

kebin0591 committed:
Relocate OrderMapper.xml to Resources Directory

jisoo78 reviewed (commented):
dto 클래스에서 lombok을 잘 활용하신 거 같네요 저도 참고해서 작성해야겠습니다

ChaeUngLim reviewed (approved):
확인

kebin0591 committed:
feat: Implement Order Creation Flow and Enhance Cart Management

- **CartItem** 및 **CartItemDTO** 클래스를 추가하여 비즈니스 로직과 데이터 전송 객체를 분리하였습니다.
- **OrderMainMenu**에 "주문하기" 메뉴 옵션을 추가하였습니다.
  - 사용자가 도서 코드를 검색하거나 전체 도서를 조회할 수 있습니다 (임시 구현).
  - 사용자가 도서를 장바구니에 담고, 수량을 지정하며, 결제 단계로 진행할 수 있습니다.
- **OrderController**에 `createOrder` 메서드를 추가하여 주문 생성을 처리할 수 있도록 하였습니다.
- **OrderService**를 업데이트하여 여러 주문 항목을 포함한 주문 생성을 지원합니다.
- **OrderDAO**에 `insertOrderItem` 메서드를 추가하여 데이터베이스에 주문 항목을 추가할 수 있도록 하였습니다.
- **OrderMapper** 인터페이스를 수정하여 주문 항목에 대한 SQL 쿼리를 매핑하였습니다.
- **Cart** 클래스를 개선하여 장바구니 항목을 관리하고, 데이터를 전송하기 위해 **CartDTO**로 변환할 수 있도록 하였습니다.
- **Input** 및 **Menu** 클래스에 유틸리티 메서드를 추가하여 사용자 상호작용을 개선하였습니다.

kebin0591 committed:
feat: Add menu options for ordering books in OrderMainMenu

- OrderMainMenu 클래스에 주문하기 메뉴 옵션을 추가.
- 사용자는 전체 도서 목록을 조회하거나 도서 코드를 통해 검색 가능.
- utility를 사용한 리팩토링

ChaeUngLim reviewed (changes requested):
mapper에서 id="deleteOrder" parameterType="_int"   int에서 _int로 변경하면 좋은 것 같습니다.
